### ISSUE TITLE
The accuracy here refers to the training accuracy or the test accuracy？

### ISSUE THREAD

**Issue by Icespellcaster** [google-research/vision_transformer#8]
Hello, does the accuracy mentioned below refer to the highest training accuracy achieved during the training process, or the test accuracy obtained by applying the trained model to the test set? Looking forward to your reply.

**Comment by andsteing** [comment]
It refers to the final (not highest) test accuracy after 10k steps (cifar10, cifar100) / 20k steps (imagenet2012)

**Comment by Icespellcaster** [comment]
> 
> 
> It refers to the final (not highest) test accuracy after 10k steps (cifar10, cifar100) / 20k steps (imagenet2012)

What is the training accuracy after training? Is the training accuracy significantly higher than the test accuracy? For example, can the training accuracy reach 98%? Looking forward to your reply.

**Comment by andsteing** [comment]
Yes, the training accuracy is significantly higher. For example a L/32 model pretrained on imagenet21k and finetuned on cifar100 achieves an accuracy >99% on the train set (compared to ~93% on the test set).

Note that you can train any large model for a long time on the train set to achieve (near) perfect accuracy, as long as you have a large enough capacity, so this metric is not very meaningful.

**Comment by cissoidx** [comment]
The test accuracy you mentioned is the split part of the training dataset, right? not the accuracy on the 12G test.tar dataset?
looking forward to your reply.



> Yes, the training accuracy is significantly higher. For example a L/32 model pretrained on imagenet21k and finetuned on cifar100 achieves an accuracy >99% on the train set (compared to ~93% on the test set).
> 
> Note that you can train any large model for a long time on the train set to achieve (near) perfect accuracy, as long as you have a large enough capacity, so this metric is not very meaningful.

**Comment by andsteing** [comment]
We're reporting numbers on the `validation` split for `imagenet2012` and on the `test` split for `cifar10` and `cifar100` (split names referring to `tensorflow_datasets` - see their [catalog](https://www.tensorflow.org/datasets/catalog/overview)):

https://github.com/google-research/vision_transformer/blob/a51dbfa231262ffd1e8e52c041324e6de644d99b/vit_jax/input_pipeline.py#L32-L54

**Comment by cissoidx** [comment]
imagenet train[:99%] means it is only using 99% of the training data?

**Comment by andsteing** [comment]
Yes it means it's using 99% of the `train` split.

**Comment by cissoidx** [comment]
Thanks for your reply.

but why not use 100% of the ILSVRC2012_img_train.tar to train, while there is actually a ILSVRC2012_img_val.tar? 

In the cifar case, I understand that there is not an official validation dataset, so we take 2% to validate the training.

However, in the imagenet2012 case, the default is to take 1% of ILSVRC2012_img_train.tar, and merge it with ILSVRC2012_img_val.tar, and then use the merged dataset to validate training process?

**Comment by cissoidx** [comment]
> Yes it means it's using 99% of the `train` split.

Hello @andsteing ,
Thanks for your reply. I tried to reproduce your results on README, but the accuracy is always about 0.5% lower than your results. I would like to confirm that the accuracy numbers on README is obtained using only 99% of the training data. Could you please confirm it?

cheers,

**Comment by andsteing** [comment]
Do you use `config.transformer.dropout_rate=0.0` ?

**Comment by cissoidx** [comment]
> Do you use `config.transformer.dropout_rate=0.0` ?

No, i was using default setting. and I found your comment in another issue saying that dropout=0 on TPU should get 84.63%. I will try with this setting. Thanks.

I will still very curious why you use only 99% of ILSVRC2012_img_train.tar for training, while there is a public validation dataset. Does this mean that if I use 100% of ILSVRC2012_img_train.tar, I can get better results on validation dataset? My work is to get comparable results between models, and this could be very important. Could you please confirm? Thanks.

cheers,

**Comment by andsteing** [comment]
Yeah sorry about the table reporting the values with `dropout=0` (note though that there is a comment just above the table) ... We will fix that soon with a new version where we sweep over the dropouts and then also include best configs in the code.

`imagenet2012` does indeed have a `validation` split but we're already using that one for computing the `test` metrics (because there is no public `test` split):

https://github.com/google-research/vision_transformer/blob/a51dbfa231262ffd1e8e52c041324e6de644d99b/vit_jax/input_pipeline.py#L47-L53

So we need a separate split for parameter fine-tuning etc and that's what we use the 1% split for. If you don't need a validation set you could try training over the 100% though I doubt you'll see a significant change.

Best, Andreas

**Comment by cissoidx** [comment]
> Yeah sorry about the table reporting the values with `dropout=0` (note though that there is a comment just above the table) ... We will fix that soon with a new version where we sweep over the dropouts and then also include best configs in the code.
> 
> `imagenet2012` does indeed have a `validation` split but we're already using that one for computing the `test` metrics (because there is no public `test` split):
> 
> https://github.com/google-research/vision_transformer/blob/a51dbfa231262ffd1e8e52c041324e6de644d99b/vit_jax/input_pipeline.py#L47-L53
> 
> So we need a separate split for parameter fine-tuning etc and that's what we use the 1% split for. If you don't need a validation set you could try training over the 100% though I doubt you'll see a significant change.
> 
> Best, Andreas

Thanks for your reply. 

It is confusing that you say `So we need a separate split for parameter fine-tuning etc and that's what we use the 1% split for.`

The `train[:99%]` marks that 99%  of ILSVRC2012_img_train.tar is used for training, but where did the remaining 1% go? In the code, I found ds_train (which is the 99% ILSVRC2012_img_train.tar) and ds_test (which is ILSVRC2012_img_val.tar). 

The 1% ILSVRC2012_img_train.tar is totally discarded and not used in whatever way, or it is merged with ILSVRC2012_img_val.tar in ds_test?

cheers,

**Comment by andsteing** [comment]
In the code on this repo we use `ds_train` for training and `ds_test` (which is the `validation` split) for reported "test" metrics.

In our previous parameter tuning (sweeps not part of this repo) we used the 1% for validation set. We decided to use the same training set here that we used in the previous sweeps. So the 1% train set data is not used in this repo.

**Comment by cissoidx** [comment]
Thanks!
all the best.
